gui/packages/ubuntupro/lib/pages/landscape/landscape_model.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '/core/agent_api_client.dart';
+
+enum LandscapeConfigType { manual, file }
+enum FileError { notFound, tooLarge, empty, none }
+
+class LandscapeModel extends ChangeNotifier {
+  LandscapeModel(this.client);
+  final AgentApiClient client;
+
+  LandscapeConfigType _selected = LandscapeConfigType.manual;
+
+  String _path = '';
+
+  String _fqdn = '';
+  String accountName = 'standalone';
+  String key = '';
+
+  bool _fqdnError = false;
+  FileError _fileError = FileError.none;
+
+  bool get fqdnError => _fqdnError;
+  FileError get fileError => _fileError;
+
+  set fqdn(String value) {
+    if (value.isNotEmpty && !value.startsWith('https://')) {
+      value = 'https://$value';
+    }
+    _fqdn = value;
+    validateFQDN();
+  }
+
+  String get fqdn => _fqdn;
+
+  set selected(LandscapeConfigType value) {
+    _selected = value;
+    _fqdnError = false;
+    _fileError = FileError.none;
+    validConfig();
+    notifyListeners();
+  }
+
+  LandscapeConfigType get selected {
+    return _selected;
+  }
+
+  set path(String value) {
+    _path = value;
+    validatePath();
+  }
+
+  String get path => _path;
+
+  bool validateFQDN() {
+    _fqdnError = _fqdn.isEmpty || Uri.tryParse(_fqdn) == null;
+    notifyListeners();
+    return !fqdnError;
+  }
+
+  bool validatePath() {
+    final file = File(_path);
+    final fileStat = file.statSync();
+    if (_path.isEmpty) {
+      _fileError = FileError.empty;
+    } else if (fileStat.size >= 1024 * 1024) {
+      _fileError = FileError.tooLarge;
+    } else if (fileStat.type == FileSystemEntityType.notFound) {
+      _fileError = FileError.notFound;
+    } else {
+      _fileError = FileError.none;
+    }
+    notifyListeners();
+    return fileError == FileError.none;
+  }
+
+  bool validConfig() {
+    var valid = false;
+    switch (selected) {
+      case LandscapeConfigType.manual:
+        valid = validateFQDN();
+      case LandscapeConfigType.file:
+        valid = validatePath();
+      default:
+        throw UnimplementedError('Unknown configuration type');
+    }
+
+    return valid;
+  }
+
+  Future<bool> applyConfig() async {
+    if (!validConfig()) {
+      return false;
+    }
+
+    switch (selected) {
+      case LandscapeConfigType.manual:
+        await applyManualLandscapeConfig();
+      case LandscapeConfigType.file:
+        await applyLandscapeConfig();
+      default:
+        throw UnimplementedError('Unknown configuration status');
+    }
+
+    return true;
+  }
+
+  Future<void> applyLandscapeConfig() async {
+    final file = File(_path);
+    final content = await file.readAsString();
+    await client.applyLandscapeConfig(content);
+  }
+
+  Future<void> applyManualLandscapeConfig() async {
+    final config = '''
+[host]
+url = $_fqdn:6554
```

Comment:
On the `host.url` line only we need to strip the `https://` out of the `_fqdn` per the experience we've had during the recent tests. Something like:

```suggestion
  Future<void> applyManualLandscapeConfig() async {
  final Uri uri = Uri.parse(_fqdn).replace(port: 6554);
    final config = '''
[host]
url = ${uri.authority}
```

Ref: https://api.flutter.dev/flutter/dart-core/Uri/authority.html